beluga_ros/include/beluga_ros/laser_scan.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_LASER_SCAN_HPP
+#define BELUGA_ROS_LASER_SCAN_HPP
+
+#include <range/v3/view/iota.hpp>
+
+#include <beluga/sensor/data/laser_scan.hpp>
+#include <beluga/views/take_evenly.hpp>
+#include <beluga_ros/messages.hpp>
+
+#include <sophus/se3.hpp>
+
+namespace beluga_ros {
+
+class LaserScan : public beluga::BaseLaserScan<LaserScan> {
+ public:
+  using Scalar = double;
+
+  explicit LaserScan(
+      beluga_ros::msg::LaserScanConstSharedPtr scan,
+      Sophus::SE3d origin = Sophus::SE3d{},
+      std::size_t max_beams = std::numeric_limits<std::size_t>::max(),
+      Scalar min_range = std::numeric_limits<Scalar>::min(),
+      Scalar max_range = std::numeric_limits<Scalar>::max())
```

Comment:
Do we need to make them optional, since we use them in a single place and we use them all?

Reply:
I think that default values are not unreasonable.

Reply:
I think the default values are fine. My point was that whether these parameters need to have a default value at all, since for our use case we do provide values for them.

Anyway, I don't feel too strongly about this.